components/ConfirmationModal.js

```diff
@@ -0,0 +1,40 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import StyledButton from './StyledButton';
+import Container from './Container';
+import { P } from './Text';
+import Modal, { ModalBody, ModalHeader, ModalFooter } from './StyledModal';
+
+const ConfirmationModal = ({ show, header, body, onClose, actions }) => (
+  <Modal width="570px" show={show} onClose={() => onClose()}>
+    <ModalHeader>{header}</ModalHeader>
+    <ModalBody>
+      <P>{body}</P>
+    </ModalBody>
+    <ModalFooter>
+      <Container display="flex" justifyContent="flex-end">
+        <StyledButton mx={20} onClick={() => actions.cancelHandler()}>
+          {actions.cancelText}
+        </StyledButton>
+        <StyledButton data-cy="action" buttonStyle="primary" onClick={() => actions.continueHandler()}>
+          {actions.continueText}
+        </StyledButton>
+      </Container>
+    </ModalFooter>
+  </Modal>
+);
+
+ConfirmationModal.propTypes = {
+  show: PropTypes.bool.isRequired,
+  header: PropTypes.string.isRequired,
+  body: PropTypes.string.isRequired,
+  onClose: PropTypes.func.isRequired,
+  actions: PropTypes.shape({
```

Comment:
What's the rational for passing an `actions` object rather that individual props?

Reply:
Technically no reason, just thought about breaking the props into categories and a lot of individual props doesn't really look cool to me.

Reply:
In practice it takes more lines to pass as an object than as individual props (6 vs 4). So I don't really see the code style argument.

It also makes some simple things more complex, for example one could expect this component to provide default labels for the actions (because most of the time labels will be `Cancel` / `Confirm`). If we want to add that at some point we'll have to merge the `actions` object with a `defaultActions` object, whereas it just requires to add an entry to `defaultProps` when props are passed as a single items (and `defaultProps` are also automatically imported in styleguide docs). 

In technical terms there's also one potential disadvantage to do that: because the `actions` object is generated during the render a new one will be created each time. If we need to optimize the `ConfirmationModal` at some point it will prevent us to use `React.memo` of `React.PureComponent` because the shallow comparison will always be false even if the content doesn't change (`{} === {} // false`). 

I'm being very picky on this one, but because this component is meant to be widely re-used it's cool if we can have the best API possible :slightly_smiling_face: 

![image](https://user-images.githubusercontent.com/1556356/67299403-edc3c480-f4ec-11e9-8969-ceda4dc9bdea.png)

Reply:
I like the idea of setting default action label, you're right individual props make things simpler and I'm up for simplicity. I'll update this